core/service.go

```diff
@@ -41,26 +40,28 @@ var _ services.Service = &Service{}
 // BABE session, and p2p service. It deals with the validation of transactions
 // and blocks by calling their respective validation functions in the runtime.
 type Service struct {
-	blockState   BlockState
-	storageState StorageState
-	rt           *runtime.Runtime
-	bs           *babe.Session
-	keys         []crypto.Keypair
-	blkRec       <-chan types.Block // receive blocks from BABE session
-	msgRec       <-chan p2p.Message // receive messages from p2p service
-	epochDone    <-chan struct{}    // receive from this channel when BABE epoch changes
-	msgSend      chan<- p2p.Message // send messages to p2p service
+	blockState    BlockState
+	storageState  StorageState
+	rt            *runtime.Runtime
+	bs            *babe.Session
+	keys          []crypto.Keypair
+	blkRec        <-chan types.Block // receive blocks from BABE session
+	msgRec        <-chan p2p.Message // receive messages from p2p service
+	epochDone     <-chan struct{}    // receive from this channel when BABE epoch changes
+	msgSend       chan<- p2p.Message // send messages to p2p service
+	babeAuthority bool
```

Comment:
Maybe `isAuthority` ?

Reply:
hmm, we might have an option for grandpaAuthority in the future, so I wanted to be a bit more specific. maybe `isBabeAuthority`?

Reply:
Gotcha. Yea, maybe `isBabeAuthority` to imply a bool...? Might not be necessary though.

Reply:
agree with the pattern is before the bool.

Reply:
will update